metron-platform/metron-common/src/test/java/org/apache/metron/common/stellar/StellarTest.java

```diff
@@ -0,0 +1,422 @@
+  }
+
+  @Test
+  public void testURLToPort() {
+    String query = "URL_TO_PORT(foo)";
+    Assert.assertEquals(80, run(query, ImmutableMap.of("foo", "http://www.google.co.uk/my/path")));
+  }
+
+  @Test
+  public void testURLToProtocol() {
+    String query = "URL_TO_PROTOCOL(foo)";
+    Assert.assertEquals("http", run(query, ImmutableMap.of("foo", "http://www.google.co.uk/my/path")));
+  }
+
+  @Test
+  public void testURLToPath() {
+    String query = "URL_TO_PATH(foo)";
+    Assert.assertEquals("/my/path", run(query, ImmutableMap.of("foo", "http://www.google.co.uk/my/path")));
+  }
+
+  @Test
+  public void testProtocolToName() {
+    String query = "PROTOCOL_TO_NAME(protocol)";
+    Assert.assertEquals("TCP", run(query, ImmutableMap.of("protocol", "6")));
+    Assert.assertEquals("TCP", run(query, ImmutableMap.of("protocol", 6)));
+    Assert.assertEquals(null, run(query, ImmutableMap.of("foo", 6)));
+    Assert.assertEquals("chicken", run(query, ImmutableMap.of("protocol", "chicken")));
+  }
+  @Test
+  public void testDateConversion() {
+    long expected =1452013350000L;
+    {
+      String query = "TO_EPOCH_TIMESTAMP(foo, 'yyyy-MM-dd HH:mm:ss', 'UTC')";
+      Assert.assertEquals(expected, run(query, ImmutableMap.of("foo", "2016-01-05 17:02:30")));
+    }
+    {
+      String query = "TO_EPOCH_TIMESTAMP(foo, 'yyyy-MM-dd HH:mm:ss')";
+      Long ts = (Long) run(query, ImmutableMap.of("foo", "2016-01-05 17:02:30"));
+      //is it within 24 hours of the UTC?
+      Assert.assertTrue(Math.abs(ts - expected) < 8.64e+7);
+    }
+  }
+
+  @Test
+  public void testToString() {
+    Assert.assertEquals("5", run("TO_STRING(foo)", ImmutableMap.of("foo", 5)));
+  }
+
+  @Test
+  public void testToInteger() {
+    Assert.assertEquals(5, run("TO_INTEGER(foo)", ImmutableMap.of("foo", "5")));
+    Assert.assertEquals(5, run("TO_INTEGER(foo)", ImmutableMap.of("foo", 5)));
+  }
+
+  @Test
+  public void testToDouble() {
+    Assert.assertEquals(new Double(5.1), run("TO_DOUBLE(foo)", ImmutableMap.of("foo", 5.1d)));
+    Assert.assertEquals(new Double(5.1), run("TO_DOUBLE(foo)", ImmutableMap.of("foo", "5.1")));
+  }
+  @Test
+  public void testGet() {
+    Map<String, Object> variables = ImmutableMap.of("foo", "www.google.co.uk");
+    Assert.assertEquals("www", run("GET_FIRST(SPLIT(DOMAIN_REMOVE_TLD(foo), '.'))", variables));
+    Assert.assertEquals("www", run("GET(SPLIT(DOMAIN_REMOVE_TLD(foo), '.'), 0)", variables));
+    Assert.assertEquals("google", run("GET_LAST(SPLIT(DOMAIN_REMOVE_TLD(foo), '.'))", variables));
+    Assert.assertEquals("google", run("GET(SPLIT(DOMAIN_REMOVE_TLD(foo), '.'), 1)", variables));
+  }
+  private static Object run(String rule, Map<String, Object> variables) {
+    StellarProcessor processor = new StellarProcessor();
+    Assert.assertTrue(rule + " not valid.", processor.validate(rule));
+    return processor.parse(rule, x -> variables.get(x));
+  }@Test
```

Comment:
Easy enough to have Intellij fix the spacing?  Sometimes we have spaces between methods, sometimes not, here there is no line break at all.  All simple, but add up to making it more difficult to read than it might need to be.